src/pudl/analysis/record_linkage/eia_ferc1_record_linkage.py

```diff
@@ -211,29 +212,30 @@ def get_training_data_df(inputs):
 @op
 def get_model_predictions(eia_df, ferc_df, train_df, experiment_tracker):
     """Train splink model and output predicted matches."""
-    settings_dict = {
-        "link_type": "link_only",
-        "unique_id_column_name": "record_id",
-        "additional_columns_to_retain": ["plant_id_pudl", "utility_id_pudl"],
-        "comparisons": COMPARISONS,
-        "blocking_rules_to_generate_predictions": BLOCKING_RULES,
-        "retain_matching_columns": True,
-        "retain_intermediate_calculation_columns": True,
-        "probability_two_random_records_match": 1 / len(eia_df),
-    }
-    linker = DuckDBLinker(
+    settings = SettingsCreator(
+        link_type="link_only",
+        unique_id_column_name="record_id",
+        additional_columns_to_retain=["plant_id_pudl", "utility_id_pudl"],
+        comparisons=COMPARISONS,
+        blocking_rules_to_generate_predictions=BLOCKING_RULES,
+        retain_matching_columns=True,
+        retain_intermediate_calculation_columns=True,
+        probability_two_random_records_match=(1.0 / len(eia_df)),
+    )
+    linker = Linker(
         [eia_df, ferc_df],
+        settings=settings,
         input_table_aliases=["eia_df", "ferc_df"],
-        settings_dict=settings_dict,
+        db_api=DuckDBAPI(),
```

Comment:
The `Linker` class is agnostic as to what database backend it uses -- you pass in a class that is used to interface with it.